cmake/inputs/DownloadPrebuiltTileDB.cmake

```diff
@@ -0,0 +1,137 @@
+#
+# FindTileDB_EP.cmake
+#
+#
+# The MIT License
+#
+# Copyright (c) 2023 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+function(fetch_tiledb_release_list VERSION EXPECTED_HASH)
+        # Local constants
+        set(UPSTREAM_URL "https://github.com/TileDB-Inc/TileDB/releases/download")
+
+        if(${EXPECTED_HASH})
+                file(DOWNLOAD
+                        ${UPSTREAM_URL}/${VERSION}/rellist.csv
+                        rellist.csv
+                        SHOW_PROGRESS
+                        EXPECTED_HASH ${EXPECTED_HASH}
+                        )
+        else()
+                message(WARNING "Downloading release list without SHA checksum!")
+                file(DOWNLOAD
+                        ${UPSTREAM_URL}/${VERSION}/rellist.csv
+                        rellist.csv
+                        SHOW_PROGRESS
+                        )
+        endif()
+
+        file(STRINGS
+                ${CMAKE_CURRENT_BINARY_DIR}/rellist.csv
+                RELLIST
+                )
+
+        # Remove csv table headers
+        list(POP_FRONT RELLIST)
+
+        foreach(LINE ${RELLIST})
+                string(REPLACE "," ";" LINE ${LINE})
+                list(LENGTH LINE LENGTH)
+
+                list(GET LINE 0 OS)
+                list(GET LINE 1 ARCH)
+                list(GET LINE 2 URL)
+                list(GET LINE 3 SHA)
+
+                if(${LENGTH} GREATER 4)
+                        list(GET LINE 4 NOAVX2)
+                endif()
```

Comment:
I still have some reservations on how good this schema is. A more scalable and clean solution would be a `PLATFORM,URL,SHA`, where `PLATFORM` would be a string like `x64-windows`, `x64-linux-noavx2`, or `source`.